keps/sig-scheduling/2044-topology-awareness-in-kube-scheduler/README.md

```diff
@@ -0,0 +1,767 @@
+  Manager as part of the filter extension point to filter out nodes that are not suitable for the workload based on the resource request and the obtained 
+  NodeResourceTopology information corresponding to that worker node. The scoring extension point to determine a score based on a configurable strategy
+  would be done as a Beta feature. 
+- The scheduler plugin `NodeResourceTopologyMatch` would be disabled by default and when enabled would also check for the ability to run pod only
+  in case of Topology Manager policy of single-numa-node policy is configured on the node. Since it is the most strict policy, it implies that the launch
+  on the node with other existing policies will be successful if the condition for single-numa-node policy passed for the worker node.
+ 
+
+### Design Consideration
+
+By default Kubelet would be resposible to fill the NodeResourceTopology instances, but a design consideration is to architect the
+solution in such a way to allow future extension to facilitate the cloud administratore to disable the feature and introduce a custom external agents to fill the NodeResourceTopology data.
+
+### Changes to the API
+
+Code responsible for working with NodeResourceTopology API will be placed in the staging directory
+at path staging/src/k8s.io/api/node/v1/types.go.
+
+```go
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeResourceTopology is a specification for a hardware resources
+type NodeResourceTopology struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	TopologyPolicies []string 		   `json:"topologyPolicies"`
+	Cells            map[string]Cell   `json:"cells"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeResourceTopologyList is a list of NodeResourceTopology resources
+type NodeResourceTopologyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []NodeResourceTopology `json:"items"`
+}
+
+// Cell is the spec for a NodeResourceTopology resource
+type Cell struct {
+    Name       string           `json:"name"`
+    Type       string           `json:"type"`
+    Parent     string           `json:"parent,omitempty"`
+    Costs      []CostInfo       `json:"costs,omitempty"`
+    Attributes []AttributeInfo  `json:"attributes,omitempty"`
+    Resources  []ResourceInfo   `json:"resources,omitempty"`
+}
+
+type ResourceInfo struct {
+    Name        string `json:"name"`
+    Allocatable string `json:"allocatable"`
+    Capacity    string `json:"capacity"`
+}
+
+type CostInfo struct {
+    Name  string `json:"name"`
+    Value int	 `json:"value"`
+}
+
+type AttributeInfo struct {
+    Name  string `json:"name"`
+    Value string `json:"value"`
+}
+// Kubelet writes to NodeResourceTopology
```

Comment:
@rphillips Here I have provided real world example of how these fields would look like once populated. Thought it was best to do that after all the data structures have been defined. Would you prefer for it to be moved before the data structures are defined or maybe in a different section?